File: Bronze/1085/1085.cpp
// [1085] 직사각형에서 탈출하기
#include <iostream>
using namespace std;

int main(void)
{
    int x, y, w, h, min;
    cin >> x >> y >> w >> h;
    
    min = (w-x) < x ? (w-x) : x;
    min = min < y ? min : y;
    min = min < (h-y) ? min : (h-y);
    
    cout << min << "\n";
}

File: Bronze/1085/1085.java
// [1085] 직사각형에서 탈출하기
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class Main {
    public static void main(String[] args) throws IOException
    {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        String[] s = br.readLine().split(" ");
        int x = Integer.parseInt(s[0]);
        int y = Integer.parseInt(s[1]);
        int w = Integer.parseInt(s[2]);
        int h = Integer.parseInt(s[3]);

        int min = Math.min((w-x), x);
        min = Math.min(min, y);
        min = Math.min(min, (h-y));

        System.out.println(min);
    }
}